### DISCUSSION TITLE
sudo problem in linuxbrew

### DISCUSSION THREAD

**Discussion by emninger** [Homebrew/discussions#669] [Linux]
I installed linuxbrew within the default linux container of Chrome OS (which is ubuntu or debian). Now, there i have a problem with sudo. Not any command installed by linuxbrew (inxi, htop, pacapt ...) i try to run as sudo works. I always get the error:

`<user>@penguin:~$ sudo htop`
`-bash: sudo: command not found`
(without sudo, htop works)

I'd presume it's a problem regarding the environment settings ... (?)

I also tried the command "sudo -E htop"  (from man sudo) but with no success.

TIA for any pointer

**Comment by SMillerDev** [comment]
What does `sudo echo $PATH` show you?

**Reply by emninger** [reply]
`-bash: sudo: command not found`

**Reply by SMillerDev** [reply]
So you can't run sudo because there is no sudo. That doesn't really seem like a linuxbrew problem?

**Comment by sroby** [comment]
Never used Chrome OS but I assume this means it doesn't include sudo by default?

**Reply by emninger** [reply]
I think ifound out: 
I added via visudo in the sudoers file the /home/linuxbrew/.linuxbrew ... directories

**Reply by gromgit** [reply]
Just to confirm: Does `sudo echo $PATH` now work for you?

**Reply by emninger** [reply]
Yes!

`/home/xxx/perl5/bin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/home/linuxbrew/.linuxbrew/bin`

**Reply by gromgit** [reply]
Then it wasn't:
> I added via visudo in the sudoers file the /home/linuxbrew/.linuxbrew ... directories

that fixed it. Your original error:
> `-bash: sudo: command not found`

was *bash* complaining that it couldn't find the `sudo` binary, while what you did would only affect what happens *after* bash is able to find (and run) `sudo`.

**Reply by emninger** [reply]
hmm ... ? Seems logical to me, but i didn't do anything else, effectively. 'sudo htop' eg. works like a charm now.

My impression was, that the complaint re sudo command not found was related to the complete command, i.e. 'sudo htop' (in this case) and not to only sudo; because, i could do 'sudo su -' to gain root even before changing the sudoers file .... (?)

**Reply by gromgit** [reply]
> My impression was, that the complaint re sudo command not found was related to the complete command, i.e. 'sudo htop' (in this case) and not to only sudo

That's not how bash (or zsh, fish, and every other *nix command shell) works. Only the first word in a command invocation is used to resolve the program to be executed; the other words are passed as arguments to that program. `sudo htop` tells your shell to:
1. find the `sudo` program, then
2. run it with a single argument: `htop`

The error message you got says step 1 failed.

DId you open a new terminal window, or start a new shell, after which sudo started working again in that new window/shell? If so, your previous shell session likely had a broken `PATH`.
